lib/resources/forms.js

```diff
@@ -137,7 +137,7 @@ module.exports = (service, endpoint) => {
           : getPartial(Forms, request, project, Keys)))
         .then((partial) => Promise.all([
           Forms.createVersion(partial, form, false),
-          Submissions.clearDraftSubmissions(form.id)
+          Submissions.softDeleteDraftSubmissions(form.id)
         ]))
         .then(() => Forms.clearUnneededDrafts(form))) // remove drafts made obsolete by new draft
```

Comment:
Now that this endpoint will no longer result in a new form draft that's ready to be purged, I think we can remove this line. It'd be a nice way to simplify the endpoint. I also think it'd make things clearer if there was only one thing that called `clearUnneededDrafts()` (namely, the purge task).

Reply:
One possible reason to leave it here is when there are no submissions, the draft def can be cleaned up right away. There are some tests in "purging form fields of unneeded drafts" (in `api: /projects/:id/forms (drafts)`) that are checking for these defs to be purged immediately. I could change or remove those tests, though.